Hmm I might be confused how to fix this test, it just generically doesn't seem to like conversion:

```
E           AssertionError: assert {'selection':...gitude': 1.0}} == {'selection':...tude': '1.0'}}                                                            x _ Webpack
E             Differing items:                                                                                                                                 x   Compiled successfully in 2.51s
E             {'selection': {'latitude': 1.0, 'longitude': 1.0}} != {'selection': {'latitude': 1.0, 'longitude': '1.0'}}        
```

--- a/tests/helpers/test_selector.py
+++ b/tests/helpers/test_selector.py
@@ -858,7 +858,6 @@ def test_language_selector_schema(schema, valid_selections, invalid_selections) 
                     "longitude": 2.0,
                     "radius": 3.0,
                 },
-                {"latitude": 1.0, "longitude": "1.0"},
                 {
                     "latitude": 1,
                     "longitude": 2,
@@ -878,18 +877,7 @@ def test_language_selector_schema(schema, valid_selections, invalid_selections) 
 def test_location_selector_schema(schema, valid_selections, invalid_selections) -> None:
     """Test location selector."""
 
-    def location_converter(x):
-        if isinstance(x, dict):
-            converted_location: dict[str, float] = {}
-            for key in x:
-                converted_location[key] = float(x[key])
-            return converted_location
-        else:
-            return x
-
-    _test_selector(
-        "location", schema, valid_selections, invalid_selections, location_converter
-    )
+    _test_selector("location", schema, valid_selections, invalid_selections)
 
 
 @pytest.mark.parametrize(

--- a/tests/helpers/test_selector.py
+++ b/tests/helpers/test_selector.py
@@ -879,9 +879,13 @@ def test_location_selector_schema(schema, valid_selections, invalid_selections) 
     """Test location selector."""
 
     def location_converter(x):
-        for key in x:
-            x[key] = float(x[key])
-        return x
+        if isinstance(x, dict):
+            converted_location: dict[str, float] = {}
+            for key in x:
+                converted_location[key] = float(x[key])
+            return converted_location
+        else:
+            return x
 
     _test_selector(
         "location", schema, valid_selections, invalid_selections, location_converter

--- a/tests/helpers/test_selector.py
+++ b/tests/helpers/test_selector.py
@@ -878,7 +878,14 @@ def test_language_selector_schema(schema, valid_selections, invalid_selections) 
 def test_location_selector_schema(schema, valid_selections, invalid_selections) -> None:
     """Test location selector."""
 
-    _test_selector("location", schema, valid_selections, invalid_selections)
+    def location_converter(x):
+        for key in x:
+            x[key] = float(x[key])
+        return x
+
+    _test_selector(
+        "location", schema, valid_selections, invalid_selections, location_converter
+    )
 
 
 @pytest.mark.parametrize(

